bool sim
---
int32[] board
